pkg/cmd/server/server.go

```diff
@@ -245,6 +246,9 @@ func (c *Config) Complete(ctx context.Context) (RunnableServer, error) {
 			// are at most the number of elements returned from a datastore query
 			datastorecfg.WithFilterMaximumIDCount(c.DispatchChunkSize),
 			datastorecfg.WithEnableRevisionHeartbeat(c.EnableRevisionHeartbeat),
+			// NOTE: this isn't something that you can meaningfully set from the outside, so we set the
+			// defaul there so that downstream references to the configuration can make use of it.
```

Comment:
this comment doesn't make sense to me. if the caller can't set this, then why offer an option at all? just set the value in the constructor

also, can we write a test that reproduces the segfault?